projects/plugins/jetpack/modules/custom-post-types/nova.php

```diff
@@ -28,1717 +28,1721 @@
 use Automattic\Jetpack\Assets;
 use Automattic\Jetpack\Roles;
 
-/**
- * Create the new Nova CPT.
- */
-class Nova_Restaurant {
-	const MENU_ITEM_POST_TYPE = 'nova_menu_item';
-	const MENU_ITEM_LABEL_TAX = 'nova_menu_item_label';
-	const MENU_TAX            = 'nova_menu';
+if ( ! class_exists( '\Nova_Restaurant' ) ) {
 
 	/**
-	 * Version number used when enqueuing all resources (css and js).
-	 *
-	 * @var string
+	 * Create the new Nova CPT.
 	 */
-	public $version = '20210303';
+	class Nova_Restaurant {
+		const MENU_ITEM_POST_TYPE = 'nova_menu_item';
+		const MENU_ITEM_LABEL_TAX = 'nova_menu_item_label';
+		const MENU_TAX            = 'nova_menu';
 
-	/**
-	 * Default markup for the menu items.
-	 *
-	 * @var array
-	 */
-	protected $default_menu_item_loop_markup = array(
-		'menu_tag'               => 'section',
-		'menu_class'             => 'menu-items',
-		'menu_header_tag'        => 'header',
-		'menu_header_class'      => 'menu-group-header',
-		'menu_title_tag'         => 'h1',
-		'menu_title_class'       => 'menu-group-title',
-		'menu_description_tag'   => 'div',
-		'menu_description_class' => 'menu-group-description',
-	);
+		/**
+		 * Version number used when enqueuing all resources (css and js).
+		 *
+		 * @var string
+		 */
+		public $version = '20210303';
 
-	/**
-	 * Array of markup for the menu items.
-	 *
-	 * @var array
-	 */
-	protected $menu_item_loop_markup = array();
+		/**
+		 * Default markup for the menu items.
+		 *
+		 * @var array
+		 */
+		protected $default_menu_item_loop_markup = array(
+			'menu_tag'               => 'section',
+			'menu_class'             => 'menu-items',
+			'menu_header_tag'        => 'header',
+			'menu_header_class'      => 'menu-group-header',
+			'menu_title_tag'         => 'h1',
+			'menu_title_class'       => 'menu-group-title',
+			'menu_description_tag'   => 'div',
+			'menu_description_class' => 'menu-group-description',
+		);
 
-	/**
-	 * Last term ID of a loop of menu items.
-	 *
-	 * @var bool|int
-	 */
-	protected $menu_item_loop_last_term_id = false;
+		/**
+		 * Array of markup for the menu items.
+		 *
+		 * @var array
+		 */
+		protected $menu_item_loop_markup = array();
 
-	/**
-	 * Current term ID of a loop of menu items.
-	 *
-	 * @var bool|int
-	 */
-	protected $menu_item_loop_current_term = false;
+		/**
+		 * Last term ID of a loop of menu items.
+		 *
+		 * @var bool|int
+		 */
+		protected $menu_item_loop_last_term_id = false;
 
-	/**
-	 * Initialize class.
-	 *
-	 * @param array $menu_item_loop_markup Array of markup for the menu items.
-	 *
-	 * @return self
-	 */
-	public static function init( $menu_item_loop_markup = array() ) {
-		static $instance = false;
+		/**
+		 * Current term ID of a loop of menu items.
+		 *
+		 * @var bool|int
```

Comment:
`menu_item_loop_current_term` seems to also be able to be of type `Wp_Term` after the assignment [here](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/modules/custom-post-types/nova.php#L1588).

Also there are usages later that could result in Warnings (see 7ffe4564fc4c2a609c7f3ecbb525f372-logstash)

Reply:
Good catch. I've updated them in the package version of the class here: dae867c35a60ccd84e2af851ccfdd3280953c3ea, makes sense to update them now before requiring the package file, if possible. It doesn't seem to make sense (to me) to fix it in `nova.php` though, given that we'll be deprecating that in a couple of weeks (plus I don't want to introduce possibly unexpected side effects just before everyone leaves for holidays).

Reply:
I am slightly contradicting myself though, as that's why I made the JS changes in the module files (to make sure issues could be caught and then updated in the package versions later), but I think I'm just trying to limit possible issues over the holidays.